### ISSUE TITLE
Incorrect future day calculation when using weekdays

### ISSUE THREAD

**Issue by undsoft** [Hexagon/croner#54]
"croner": "~4.1.96"

```
const cron = new Cron('0 0 * * sun');
const previousRun = new Date('2022-02-17');
const nextRun = cron.next(previousRun);
```

Expected: Sun Feb 20 2022 00:00:00
Actual: Sun Jan 01 2023 00:00:00

**Comment by Hexagon** [comment]
Thanks, will add test cases and have a look at that asap.

**Comment by Hexagon** [comment]
Fixed in 4.1.97
